more flaky tests: `test_log_message_filtering`

```

=================================== FAILURES ===================================
__________________________ test_log_message_filtering __________________________
[gw2] linux -- Python 3.11.14 /opt/hostedtoolcache/Python/3.11.14/x64/bin/python3

    async def test_log_message_filtering():
        """Test that warning and error log messages trigger notifications."""
        app = TextualAgent(
            model=DeterministicModel(
                outputs=[
                    "/warning Test warning message",
                    "Normal response",
                    "end: \n```bash\necho COMPLETE_TASK_AND_SUBMIT_FINAL_OUTPUT\n```",
                ]
            ),
            env=LocalEnvironment(),
            mode="yolo",
        )
    
        # Mock the notify method to capture calls
        app.notify = Mock()
    
        async with app.run_test() as pilot:
            # Start the agent with the task
            threading.Thread(target=lambda: app.agent.run("Log test"), daemon=True).start()
            await pilot.pause(0.2)
    
            # Verify warning was emitted and handled (note the extra space in the actual format)
      return self.get_observation(self.query())
                                  ^^^^^^^^^^^^
    File "/home/runner/work/mini-swe-agent/mini-swe-agent/src/minisweagent/agents/interactive_textual.py", line 60, in query
      return super().query()
             ^^^^^^^^^^^^^^^
    File "/home/runner/work/mini-swe-agent/mini-swe-agent/src/minisweagent/agents/default.py", line 98, in query
      self.add_message("assistant", **response)
    File "/home/runner/work/mini-swe-agent/mini-swe-agent/src/minisweagent/agents/interactive_textual.py", line 50, in add_message
      self.app.call_from_thread(self.app.on_message_added)
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/site-packages/textual/app.py", line 1738, in call_from_thread
      result = future.result()
               ^^^^^^^^^^^^^^^
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/concurrent/futures/_base.py", line 456, in result
      return self.__get_result()
             ^^^^^^^^^^^^^^^^^^^
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
      raise self._exception
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/site-packages/textual/app.py", line 1732, in run_callback
      return await invoke(callback_with_args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/site-packages/textual/_callback.py", line 96, in invoke
      return await _invoke(callback, *params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/site-packages/textual/_callback.py", line 56, in _invoke
      result = callback(*params[:parameter_count])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/runner/work/mini-swe-agent/mini-swe-agent/src/minisweagent/agents/interactive_textual.py", line 334, in on_message_added
      self.update_content()
    File "/home/runner/work/mini-swe-agent/mini-swe-agent/src/minisweagent/agents/interactive_textual.py", line 358, in update_content
      container = self.query_one("#content", Vertical)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/site-packages/textual/dom.py", line 1499, in query_one
      raise NoMatches(f"No nodes match {query_selector!r} on {base_node!r}")
  textual.css.query.NoMatches: No nodes match '#content' on Screen(id='_default')
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
      self.run()
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/threading.py", line 982, in run
      self._target(*self._args, **self._kwargs)
    File "/home/runner/work/mini-swe-agent/mini-swe-agent/tests/agents/test_interactive_textual.py", line 271, in <lambda>
      threading.Thread(target=lambda: app.agent.run("Log test"), daemon=True).start()
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/runner/work/mini-swe-agent/mini-swe-agent/src/minisweagent/agents/interactive_textual.py", line 67, in run
      self.app.call_from_thread(self.app.action_quit)
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/site-packages/textual/app.py", line 1735, in call_from_thread
      future: Future[CallThreadReturnType] = asyncio.run_coroutine_threadsafe(
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/asyncio/tasks.py", line 936, in run_coroutine_threadsafe
      loop.call_soon_threadsafe(callback)
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/asyncio/base_events.py", line 807, in call_soon_threadsafe
      self._check_closed()
    File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/asyncio/base_events.py", line 520, in _check_closed
Error:       raise RuntimeError('Event loop is closed')
  RuntimeError: Event loop is closed
  
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))👋 This is mini-swe-agent version 1.16.0.
Loading global config from '/home/runner/.config/mini-swe-agent/.env'


tests/agents/test_interactive_textual.py::test_list_content_rendering
  /opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/site-packages/_pytest/threadexception.py:74: RuntimeWarning: coroutine 'App.call_from_thread.<locals>.run_callback' was never awaited
    del errors, meta, hook_error
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.

tests/models/test_cache_control.py::test_set_cache_control_offset_deprecated
  /home/runner/work/mini-swe-agent/mini-swe-agent/src/minisweagent/models/utils/cache_control.py:45: UserWarning: last_n_messages_offset is deprecated and will be removed in the future. It has no effect.
    warnings.warn("last_n_messages_offset is deprecated and will be removed in the future. It has no effect.")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================ tests coverage ================================
_______________ coverage: platform linux, python 3.11.14-final-0 _______________

Coverage XML written to file coverage.xml
=========================== short test summary info ============================
Warning: sts/agents/test_interactive_textual.py::test_log_message_filtering - AssertionError: mock('[WARNING]  Test warning message', severity='warning') call not found
============ 1 failed, 348 passed, 6 warnings in 104.99s (0:01:44) =============
Error: Process completed with exit code 1.

```

--- a/tests/agents/test_interactive_textual.py
+++ b/tests/agents/test_interactive_textual.py
@@ -267,9 +267,17 @@ async def test_log_message_filtering():
     app.notify = Mock()
 
     async with app.run_test() as pilot:
+        # Wait for UI to be ready before starting the agent thread
+        await pilot.pause(0.2)
+
         # Start the agent with the task
         threading.Thread(target=lambda: app.agent.run("Log test"), daemon=True).start()
-        await pilot.pause(0.2)
+
+        # Wait for notify to be called (with timeout)
+        for _ in range(50):  # 5 sec timeout
+            await pilot.pause(0.1)
+            if app.notify.call_count > 0:
+                break
 
         # Verify warning was emitted and handled (note the extra space in the actual format)
         app.notify.assert_any_call("[WARNING]  Test warning message", severity="warning")

